### PR TITLE
fix: add signup page and remove js code

### PR DESCRIPTION
**Pull request by Leslie48197**
* fix: add signup page and remove js code
* convert js code to ts code
* add backend code for user registration

### CODE REVIEW

**Review by CheesyGamer77 — APPROVED** [review]
Excellent work, just fix some linting errors and we'll be good to merge

File: backend/src/index.ts [hunk 1]
```diff
@@ -1,14 +1,130 @@
...
 dotenv.config();
+const PORT = process.env.PORT || 3001;
 
 const app = express();
 app.use(express.json());
 app.use(cors());
 
-app.get('/ping', (_, res) => {
+// Define the MongoDB connection URL
+const mongodbUri = process.env.MONGODB_URI;
```
Comment (tfurey96) [inline backend/src/index.ts:16]: DATABASE_URL = process.env.DATABASE_URL

**Review by huan3846 — APPROVED** [review]
Nice job

**Review by tfurey96 — APPROVED** [review]
looks good some minor file structure changes may be needed in the future.

**Pull request merged**